### DISCUSSION TITLE
Add estimates to issues

### DISCUSSION THREAD

**Discussion by iosifnicolae2** [community/community#4416] [Projects and Issues] [answered]
Hi.

Would be great if you could allow developers to add estimates to each task (similar with how Zenhub it's working..).

Thanks!

**Comment by jdmcd** [comment]
I agree with this - it would be great if this was baked in as a default field and then could be summable/viewable across some kind of time period. We use linear.app right now which is really, really good at doing this, and I think that's the only feature that would prevent us from moving to GH Issues.

**Comment by petewalker** [comment]
Related issue: https://github.com/github/feedback/discussions/5376

**Comment by maddyjoe** [comment]
I think with the new Projects improvements that's in Beta atm you can add custom fields super-easily - including text or numeric. So was thinking you could add a numeric field called "Story points" and then use that for estimates? On the wait-list for Beta so once I have access will see if it can be used in that way (we've been using category tags for estimates so it appears on Kanban - but it's a chew!)

**Reply by StefanKock** [reply]
When you only use one board/project to manage tickets, that's sufficient. But with multi team setups you likely need several boards and then the estimate is to loosely coupled to the ticket or lost when it is moved between boards (or creates inconsistencies).

**Reply by maddyjoe** [reply]
yeah - so as a workaround I ended up created a "product/team" custom field and using that, coupled with another numeric estimation field to allow some things to be consistently reported, but tickets to be displayed together or separately as needed for team calls. It's ok...but would say that relying on lots of custom fields for this is a pain to maintain and set up. 
Aaaand when you're in the ticket or kanban view not all custom fields display  - so not helpful unless you don't mind constantly switching tabs. Have upvoted in case it helps get this looked at sooner!

**Reply by azenMatt** [reply]
@maddyjoe - we wanted to avoid the issue page sidebar getting too busy, but would it help here if the project meta-data was expanded by default if the issue is only in a single project?

**Comment by azenMatt** [comment, marked as answer]
Hi @iosifnicolae2 - have you tried using the number custom field for this? 

That would be our recommended way to track estimates using the new projects. 

Let me know if there are any gaps here that are stopping this from working for you

**Reply by iosifnicolae2** [reply]
Yes, this is a solution BUT the estimate is not displayed on the board view.
![image](https://user-images.githubusercontent.com/7108834/143674914-22688df6-eb84-42c1-aae2-eb274ee61a9a.png)
![image](https://user-images.githubusercontent.com/7108834/143674919-b88ec691-7a86-4a7a-b759-b2a6488d3eee.png)

**Reply by iosifnicolae2** [reply]
Also, would be very useful to be able to open a card in a modal dialog.
You can play a bit with Zenhub to understand better how it works.

**Reply by azenMatt** [reply]
@iosifnicolae2 - good news, you can customise what you can see on a card!

If you enter the view settings, the first option under configuration edits what will show up:

![image](https://user-images.githubusercontent.com/7584089/144229620-4cd0bdba-c8fa-4f72-8a5c-e2c903a7f351.png)
![image](https://user-images.githubusercontent.com/7584089/144229682-9d220b50-e90a-473b-b429-e6a7e02ad106.png)
![image](https://user-images.githubusercontent.com/7584089/144229722-cf4ad645-0dc0-49a9-8e7b-609e9904ed84.png)


Also 👍 on viewing the issue in the dialog, this is on our roadmap but isn't as simple as it might appear so keep an eye out for it early next year

**Reply by ghost** [reply]
Is there a way to sum this estimation to see the total of a groupPy query / selection ?

**Reply by maddyjoe** [reply]
oh my word I wondered where this was pulling from - as I have 2 teams and the display in Kanban was different - thank you so much!!!

**Reply by danielscatigno-ncpc** [reply]
> Is there a way to sum this estimation to see the total of a groupPy query / selection ?

Yes, it's missing a summarize option! I added custom fields for estimating the time for the issues and the time taken, but I can't see the total for a project

**Reply by Horschig** [reply]
see my reply here: https://github.com/orgs/community/discussions/4416#discussioncomment-5089519

**Reply by dsyrstad** [reply]
Perhaps not perfect, but you can get a sum of a numeric custom field using Project > Insights:

![image](https://github.com/community/community/assets/1719665/af11b9b9-5198-4448-ada9-4b10a3eb3e30)

Here's the configuration:
![image](https://github.com/community/community/assets/1719665/46ec75e7-f218-4784-95b5-5947f9781488)

**Comment by milanvanschaik** [comment]
Would be a nice addition to have the ability to fill only fibonacci numbers (for example). Could be as simple as a regex - though more user friendly would be to have some kind of templates for this. I know this should be possible with a `Single select`, though, in favour of https://github.com/github/feedback/discussions/5376 I think it's nice to keep it a numeric field.

**Comment by Horschig** [comment]
The biggest problem with this solution is that if you create a new project, the estimates are gone, because they are properties of the project, not of the issue (whereas it would make sense if they'd belong to the issue instead).

**Reply by yuanzhou** [reply]
I tested with adding a custom column in the project Table view to be used as Estimate with single value options. It's available to the individual issue page too.
<img width="330" alt="Screenshot 2023-10-10 at 11 24 24 PM" src="https://github.com/community/community/assets/195873/3114e782-f837-4cb6-807a-4f2c7d2f57f0">
<img width="401" alt="Screenshot 2023-10-10 at 11 24 32 PM" src="https://github.com/community/community/assets/195873/8c6ab3e8-a8da-4cec-b61c-e15f999bb146">

**Reply by Horschig** [reply]
Yes, it is a workaround, but the logic is still wrong to have an estimate as being part of a project instead of the issue itself. What if an issue is part of two projects? Or what if the issue gets deleted from a project because the scope changes? In the former case, you can have two different estimates and need to manually keep them in sync. In the latter case, the estimate will get deleted because it was removed from the project.

**Comment by koppor** [comment]
When one creates a new project using GitHub's UI, GitHub proposes "Kanban", too:

![image](https://github.com/community/community/assets/1366654/ea637793-f262-407f-8801-a02b960d0dd1)

This features the summarization of estimates:

![Kanban preview](https://github.githubassets.com/assets/memex-templates-kanban-light-511062823828.png)

I know that all the drawbacks ([estimates tied to project not to issue](https://github.com/orgs/community/discussions/4416#discussioncomment-7252530) and [no Fibonacci number](https://github.com/orgs/community/discussions/4416#discussioncomment-2303368)) still exist, but better than nothing. Even no plugin needed (historic reference [Browser extension to see story points in GitHub projects](https://github.com/banyan/github-story-points). @Horschig, the issue [#13](https://github.com/banyan/github-story-points/issues/13) IMHO requests what you have in mind)

**Reply by Horschig** [reply]
Thanks, handy to know. I checked and `Field Sum` is doing that in the Board view:

![image](https://github.com/community/community/assets/3985329/1d74c732-d352-4bdb-a685-a7aae45fc722)

RE the request, I don't think labels are a nice way either. It's just a hacky workaround. Estimate should rather be a first-class citizen at the same level as label.
